src/shared/components/SearchInput/SearchInput.jsx

```diff
@@ -33,24 +35,35 @@ function SearchInput({
           name.toLowerCase().includes(filterPhrase.toLowerCase()),
         );
         setFilteredItems(itemsFiltered);
+
         setSearchingItems((prevState) => !prevState);
       }, 300),
     [],
   );
 
+  const searchActionDebounced = useMemo(
+    () =>
+      debounce((filterPhrase) => {
+        searchAction(filterPhrase);
+      }, 300),
+    [searchAction],
+  );
+
   const handleInputChange = (event) => {
     const filterPhrase = event.target.value;
     setInputValue(filterPhrase);
     filterItems(items, filterPhrase);
```

Comment:
I don't think you need `filterItems` anymore since you've got connection with API search.

Reply:
Fixed